blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,88 @@
+description: "Sourcegraph 3.31 introduces UPDATE"
+tags: [blog, release]
+slug: "release/3.31"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+  - description: "Backend Code Insights GraphQL queries now support arguments `includeRepoRegex` and `excludeRepoRegex` to filter on repository names."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23256
+    category: Code Insights
+  - description: Operator documentation has been added to the Search Reference sidebar section.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23116
+    category: Search
+  - description: Syntax highlighting support for the Cue language.
+    url: https://cuelang.org
+    category: Search
+  - description: Reintroduced a revised version of the Search Types sidebar section.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23170
+    category: Search
+  - description: "Added support for [`git p4`'s `--use-client-spec` option](https://git-scm.com/docs/git-p4#Documentation/git-p4.txt---use-client-spec) can now be enabled by configuring the `p4.client` field in the Perforce integration."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23845
+    category: Admin
+  - description: "Batch Changes changesets can now be [set to published when previewing new or updated batch changes](https://docs.sourcegraph.com/batch_changes/how-tos/publishing_changesets#within-the-ui)."
+    url: https://github.com/sourcegraph/sourcegraph/issues/22912
+    category: Batch Changes
+  - description: "Code Insights drill-down filters now allow filtering insights data on the dashboard page using repo: filters."
+    url: https://github.com/sourcegraph/sourcegraph/issues/23186
+    category: Code Insights
+  - description: "Code Insights feature flag `DISABLE_CODE_INSIGHTS` environment variable has moved from the `repo-updater` service to the `worker` service. Any users of this flag will need to update their `worker` service configuration to continue using it."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23050
+    category: Repositories
+  - description: "The Docker Compose Jaeger container's `SAMPLING_STRATEGIES_FILE` now has a default value. If you are currently using a custom sampling strategies configuration, you may need to make sure your configuration is not overridden by the change when upgrading."
+    url: "https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/489"
+    category: Admin
+  - description: The copy icon displayed next to files and repositories will now copy the file or repository path. Previously, this action copied the URL to clipboard.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23390
+    category: Repositories
+  - description: "For Docker-Compose and Kubernetes users, the built-in main Postgres and codeintel databases have switched to an alpine Docker image. This requires re-indexing the entire database. This process can take up to a few hours on systems with large datasets."
```

Comment:
@iskyOS - do you know if any action is required for this to happen?

Reply:
Just talked to Geoffrey; it will happen automatically, but the CE team reached out to customers a week ago to make sure they know what steps to take to ensure it is successful.